routes/common/vm_helper.rb

```diff
@@ -31,35 +31,36 @@ def post(name)
     required_parameters << "name" << "location" if @mode == AppMode::WEB
     allowed_optional_parameters = ["size", "storage_size", "unix_user", "boot_image", "enable_ip4", "private_subnet_id"]
     request_body_params = Validation.validate_request_body(params, required_parameters, allowed_optional_parameters)
+    assemble_params = allowed_optional_parameters.filter_map { |k| [k, request_body_params[k]] if request_body_params[k] }.to_h
```

Comment:
```suggestion
    assemble_params = request_body_params.slice(*allowed_optional_parameters).compact
```

I think that does the right thing/